### PR TITLE
Add centralized logger.

### PR DESCRIPTION
**Pull request by jverswijver**
Adds a package level logger for the datajoint package.

### Should accomplish the following:
- [x] bring all the loggers under the same logger, in the current DJ we make loggers per module.
- [x] convert all warnings to logs with level WARNING 
- [x] update any tests
- [x] convert any applicable print statements to logs
~~- [ ] Evaluate log DB table to see if logging needs to be added~~

### Additional functionality requested by data science team:
~~- [ ] log warnings when user calls populate on empty `key_source`~~
~~- [ ] debug level logging for any insert/delete~~
- [X] example for adding handler to the logger so they can send log output to sources other than stdout

### Example for capturing the log output but maintaining dj logging
See the converted test `test_limit_warning` in the PR.
Basically you add a handler to the logger that sends all logs to a StringIO object and then retrieve the string from the object, then you remove the handler.

### CODE REVIEW

**Comment by ttngu207** [comment]
Suggesting the logging related to transaction to be set at DEBUG level
https://github.com/datajoint/datajoint-python/blob/0ff34f283c8c54c8510ee2e0b65d134fb8321efd/datajoint/connection.py#L383
https://github.com/datajoint/datajoint-python/blob/0ff34f283c8c54c8510ee2e0b65d134fb8321efd/datajoint/connection.py#L391
https://github.com/datajoint/datajoint-python/blob/0ff34f283c8c54c8510ee2e0b65d134fb8321efd/datajoint/connection.py#L400

File: datajoint/logging.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        )
+    else:
+        logger.error(f"Uncaught exception: {exc_value}")
+
+
+sys.excepthook = excepthook
+
+
+# https://github.com/tqdm/tqdm/issues/313#issuecomment-267959111
+class TqdmToLogger(io.StringIO):
```
Comment (dimitri-yatsenko) [inline datajoint/logging.py:38]: Where is this used? Does TQDM need to output anything to logger?

Comment (guzman-raphael) [reply datajoint/logging.py:38]: Agree that we actually don't need this. We don't need to manage `tqdm`'s logs.

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@jverswijver Great work! :muscle: 

Mostly ready just offering further improvements.

**Review by dimitri-yatsenko — APPROVED** [review]

**Review by guzman-raphael — APPROVED** [review]
@jverswijver Nice work! :handshake:

File: datajoint/autopopulate.py [hunk 2]
```diff
@@ -288,6 +288,9 @@ def _populate1(
                         exception=error.__class__.__name__,
                         msg=": " + str(error) if str(error) else "",
                     )
+                    logger.debug(
+                        f"Error making {key} -> {self.target.table_name} - {error_message}"
```
Comment (dimitri-yatsenko) [inline datajoint/autopopulate.py:292]: I think `table_name` does not include the schema name whereas `full_table_name` does. 
```suggestion
                        f"Error making {key} -> {self.target.full_table_name} - {error_message}"
```
